packages/schematics/angular/utility/clone.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { deepCopy } from '@angular-devkit/core';
+
+
+/**
+ * Copy a source with the supplied properties.
+ * @param source Source of the copy.
+ * @param properties Properties to copy.
+ */
+export function copyWith<T>(source: T, properties: string[]): T {
```

Comment:
https://stackblitz.com/edit/pick - Here's a version I just wrote that's properly typed.